### PR TITLE
Horizons unit tests

### PR DESCRIPTION
**Pull request by Roxeena**
Added some unit tests related to the Horizons data request feature. Also added a function to do the data requests without the use of Qt.

### CODE REVIEW

File: tests/test_horizons.cpp [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+    }
+}
+
+void testReadingHorizons(HorizonsType type, std::filesystem::path filePath,
+    double t0, double x0, double y0, double z0,
+    double t1, double x1, double y1, double z1,
+    double t2, double x2, double y2, double z2)
+{
+    // Get files and make sure they exist
+    std::filesystem::path kernel = absPath("${BASE}/tests/horizonsTest/naif0012.tls");
```
[the remark this answers is not in the document]
Comment (alexanderbock) [reply tests/test_horizons.cpp:104]: Same a few more times below

File: tests/test_horizons.cpp [hunk 2]
```diff
@@ -0,0 +1,993 @@
...
+    LERRORC("test_horizons", "Theese tests requires the Space module to be enabled");
+    CHECK(false);
+#endif // OPENSPACE_MODULE_SPACE_ENABLED
+}
+
+// Test Horizons requests and error handling
+// Test the errors that are captured by the error field in the json result
+TEST_CASE("HorizonsFile: File size too large", "[horizonsfile]") {
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    std::string observer = "@ssb";
```
Comment (alexanderbock) [inline tests/test_horizons.cpp:159]: I think these tests would be easier to read if the values are not stored in variables, but rather directly passed into the function

File: tests/test_horizons.cpp [hunk 3]
```diff
@@ -0,0 +1,993 @@
...
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    std::filesystem::path filePath =
+        absPath("${BASE}/tests/horizonsTest/parse_multiple_observers.hrz");
+    HorizonsFile horizonsFile(filePath);
+
+    // Parse matches
+    std::vector<std::string> matches =
+        horizonsFile.parseMatches("Name", "matches", ">MATCH NAME<");
+
+    CHECK(matches.size() == 3);
```
Comment (alexanderbock) [inline tests/test_horizons.cpp:756]: Should be REQUIRE

File: tests/test_horizons.cpp [hunk 4]
```diff
@@ -0,0 +1,993 @@
...
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    std::filesystem::path filePath =
+        absPath("${BASE}/tests/horizonsTest/parse_multiple_targets.hrz");
+    HorizonsFile horizonsFile(filePath);
+
+    // Parse matches
+    std::vector<std::string> matches =
+        horizonsFile.parseMatches("Name", "matches", ">MATCH NAME<");
+
+    CHECK(matches.size() == 11);
```
Comment (alexanderbock) [inline tests/test_horizons.cpp:779]: -> REQUIRE

File: tests/test_horizons.cpp [hunk 5]
```diff
@@ -0,0 +1,993 @@
...
+    std::filesystem::path filePath =
+        absPath("${BASE}/tests/horizonsTest/parse_multiple_stations.hrz");
+    CHECK(std::filesystem::is_regular_file(filePath));
+    HorizonsFile horizonsFile(filePath);
+
+    // Parse matches
+    std::vector<std::string> matches =
+        horizonsFile.parseMatches("Observatory Name", "Multiple matching stations found");
+
+    CHECK(matches.size() == 11);
```
Comment (alexanderbock) [inline tests/test_horizons.cpp:827]: -> REQUIRE

File: tests/test_horizons.cpp [hunk 6]
```diff
@@ -0,0 +1,993 @@
...
+
+
+// Test reading of data and compare a recent request with a stored file
+TEST_CASE("HorizonsFile: Reading Vector data from request", "[horizonsfile]") {
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    HorizonsType type = HorizonsType::Vector;
+    std::filesystem::path filePathVector =
+        absPath("${BASE}/tests/horizonsTest/validVectorFile.hrz");
+
+    double t0 = 706449669.18513119;
```
Comment (alexanderbock) [inline tests/test_horizons.cpp:872]: Can be `const`

Comment (alexanderbock) [reply tests/test_horizons.cpp:872]: Same below

File: tests/test_horizons.cpp [hunk 7]
```diff
@@ -0,0 +1,993 @@
...
+// Test the errors that are captured by the error field in the json result
+TEST_CASE("HorizonsFile: File size too large", "[horizonsfile]") {
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    std::string observer = "@ssb";
+    std::string target = "-74"; // MRO
+    std::string start = "2005-08-13 00:00:00";
+    std::string stop = "2022-07-01 00:00:00";
+    std::string step = "1";
+    std::string unit = "m";
+    std::filesystem::path filePath = absPath("${BASE}/tests/horizonsTest/horizonstest_1.hrz");
```
Comment (GPayne) [inline tests/test_horizons.cpp:165]: Another idea similar to what Alex is suggesting is to create structs here, like combining time variables into a single struct: `{start, stop, step, unit}`.

**Review by GPayne — APPROVED** [review]

**Pull request merged**